real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java

```diff
@@ -107,7 +107,8 @@ public void paintComponent(Graphics g) {
 
 		g.drawImage(map, 0, 0, frameWidth, frameHeight, Color.black, null);
 
-		for	(LocTuple mTuple : locs) {
+		for (int i = 0; i < locs.size(); i++) {
```

Comment:
why?   I would prefer to keep this as a for each, it allows the compiler to make things faster. Not that it really matters in this case